README.md

```diff
@@ -333,4 +333,354 @@ Other Style Guides
 
 **[⬆ back to top](#table-of-contents)**
 
+## Destructuring
+
+  - [5.1](#5.1) <a name='5.1'></a> Use object destructuring when accessing and using multiple properties of an object.
+
+  > Why? Destructuring saves you from creating temporary references for those properties.
+
+    ```javascript
+    // bad
+    function getFullName(user) {
+      const firstName = user.firstName;
+      const lastName = user.lastName;
+
+      return `${firstName} ${lastName}`;
+    }
+
+    // good
+    function getFullName(obj) {
+      const { firstName, lastName } = obj;
+      return `${firstName} ${lastName}`;
+    }
+
+    // best
+    function getFullName({ firstName, lastName }) {
```

Comment:
>  it's hiding the fact that we are getting an object
Since you're destructing the object `function({foo, bar}, ...otherArguments)`, I don't feel like this.

> and it puts the object attributes in the signature of the function
I like this a lot, since you're commiting to the shape of the object in the signature, rather than potentially 50 lines below, calling `firstName` and `lastName` on a parameter, just assuming everyone knows that it has these properties.